src/components/button/button.scss

```diff
@@ -170,11 +156,13 @@ button.md-button::-moz-focus-inner {
   }
 
   .md-ripple-container {
-    border-radius: $button-border-radius;
+    border-radius: inherit;
```

Comment:
`border-radius` isn't the shape of the ripple, it's the border limit to which the ripple will reach on its container. See #9154